pupepat/utils.py

```diff
@@ -78,14 +79,55 @@ def run_sep(data, mask_threshold=None):
     background = sep.Background(np.ascontiguousarray(data), mask=np.ascontiguousarray(data > mask_threshold))
     return sep.extract(data - background, 20.0, err=np.sqrt(data), minarea=1000, deblend_cont=1.0, filter_kernel=None)
 
-def merge_pdfs(output_directory):
-    pdf_files = glob(os.path.join(output_directory, '*.pdf'))
-    pdf_files.sort()
-    pdf_writer = PdfFileWriter()
 
-    for pdf_file in pdf_files:
-        pdf_writer.appendPagesFromReader(PdfFileReader(pdf_file))
+def prettify_focdmd(demanded_focus):
+    return '{focdmd:+.0f}'.format(focdmd=demanded_focus).replace('-', 'm').replace('+', 'p')
+
+
+def merge_pdfs(output_directory, output_table, output_pdf='pupe-pat'):
+    data = ascii.read(os.path.join(output_directory, output_table))
+    data.sort(['FOCDMD', 'filename'])
+    pdf_files = np.array([os.path.join(output_directory,
+                                       '{basename}_{sourceid}.pdf'.format(basename=os.path.splitext(row['filename'])[0],
+                                                                          sourceid=row['sourceid']))
+                          for row in data])
+    for demanded_focus in np.unique(data['FOCDMD']):
+        focus_set_indexes = data['FOCDMD'] == demanded_focus
+
+        pdf_writer = PdfFileWriter()
+        for pdf_file in pdf_files[focus_set_indexes]:
+            pdf_writer.appendPagesFromReader(PdfFileReader(pdf_file))
+
+
+        output_filename = os.path.join(output_directory, '{basename}-{focdmd}-donut.pdf'.format(basename=output_pdf,
+                                                                                                focdmd=prettify_focdmd(demanded_focus)))
+        with open(output_filename, 'wb') as output_stream:
+            pdf_writer.write(output_stream)
+
+
+def estimate_scatter(a, axis=None):
+    """
+    Caclulate the inter-quartile range of *a* (scaled to normal).
+    :param a: samples for the distribution
+    :param axis: axis to calculate the scatter
+    :return: scatter
+    """
+    lower_quartile, upper_quartile = np.percentile(a, [25, 75], axis=axis)
+    return 0.741301109 * (upper_quartile - lower_quartile)
```

Comment:
Is this a known number?

Reply:
Yes. I'll add a comment with a reference.